openedx/core/djangoapps/enrollments/api.py

```diff
@@ -509,11 +509,11 @@ def is_enrollment_valid_for_proctoring(username, course_id):
         return False
 
     enrollment = _data_api().get_course_enrollment(username, str(course_id))
-    if not enrollment['is_active']:
+    if not enrollment or not enrollment['is_active']:
```

Comment:
I don't think we need to address this now, but my reading of this code indicates that staff users who're masquerading as learners will no longer be able to see the proctoring experience with this change. Shouldn't block this diff, but I suspect we'll still have work to do to restore the behavior Support & others rely on around testing the proctoring experience.

Reply:
@matthugs It's a great point you raised.
That said, I believe this code is going to be OK with the masquerading functionality you raised. The user passed in are actually either the request user, or the masquerading user. If you look at the logic in edx-proctoring where the rule is invoked, it gets the username from either the request querystring, or the request itself as the fall back. That is supporting masquerading. See https://github.com/edx/edx-proctoring/blob/897449df7ec152d24721dcd1d7357a9f941ac7b1/edx_proctoring/views.py#L324